test/functional/nested-validation.spec.ts

```diff
@@ -67,6 +66,12 @@ describe("nested validation", function () {
 
             @ValidateNested()
             mySubClasses: MySubClass[];
+
+            @ValidateNested()
+            mySubSubClasses: MySubClass[][];
+
+            @ValidateNested()
+            mySubSubSubClasses: MySubClass[][][];
```

Comment:
does it work with tuples too? `myTupleClasses: [MyClass1, MyClass2];`

Reply:
`myTupleClasses: [MyClass1, MyClass2];` is basically simple array (we already support simple arrays). But we validate only items of array. So for example this will pass:
```ts
class Tuple1 {
  @IsNumber()
  num: number;
}

class Tuple2 {
  @IsString()
  name: string;
}

class Test {

  @ValidateNested()
  tuples: [Tuple1, Tuple2];

}

const t = new Test();
t.tuples = [new Tuple1(), new Tuple1() as any];
t.tuples[0].num = 2 as any;
(t.tuples[1] as any).num = 2 as any;

console.log(validateSync(t));
```